ui/src/main/webapp/app/query/operation-chain.html

```diff
@@ -13,6 +13,11 @@
     </md-button>
 </div>
 <div layout="row" layout-align="end">
-    <md-button class="md-raised overlapping" ng-click="ctrl.resetChain($event)">Reset operation chain</md-button>
-    <md-button class="md-primary md-raised overlapping" ng-disabled="!ctrl.canExecute()" id="execute-chain" ng-click="ctrl.executeChain()">Execute operation chain</md-button>
+    <md-button class="md-raised" ng-click="ctrl.resetChain($event)">Reset operation chain</md-button>
+    <md-button class="md-primary md-raised" ng-disabled="!ctrl.canExecute()" id="execute-chain" ng-click="ctrl.executeChain()">Execute operation chain</md-button>
+    <input type="text" name="operationName" ng-model="ctrl.operationName" placeholder="Operation Name"
+    id="operationName" aria-label="Operation Name">
+    <md-button class="md-primary md-raised" ng-disabled="!ctrl.canExecute()" id="save-chain" ng-click="ctrl.saveChain()">Save operation chain</md-button>
```

Comment:
Saving as named operation should not be the primary button. Therefore should be neutral and not be placed on the far right.

I'd be tempted to move the reset button to the right to avoid cluttering that right area and stop users accidentally clicking it